hack/generator/spec/versioning.md

```diff
@@ -0,0 +1,201 @@
+
+***Issue:** Should these be invoked before or after the main interface? We need to define the sequence for predictability.*
+
+## Illustration of Conversion
+
+To illustrate the way conversions will work using `ClusterProperties` from Microsoft.ServiceFabric. This explores conversion between version v20160301 and the hub storage version, based on a later version:
+
+| Fields                 | v20160301                        | vStorage (Hub)                     | Change           | Handler      |
+| --------------------------------- | -------------------------------- | ---------------------------------- | ---------------- | ------------ |
+| AzureActiveDirectory              | *AzureActiveDirectory            | *AzureActiveDirectory              | None             | Copy value   |
+| Certificate                       | *CertificateDescription          | *CertificateDescription            | None             | Copy value   |
+| ClientCertificateCommonNames      | *[]ClientCertificateCommonName   | *[]ClientCertificateCommonName     | None             | Copy value   |
+| ClientCertificateThumbprints      | *[]ClientCertificateThumbprint   | *[]ClientCertificateThumbprint     | None             | Copy value   |
+| ClusterCodeVersion                |                                  | *string                            | New Property     | Skip         |
+| DiagnosticsStorageAccountConfig   | *DiagnosticsStorageAccountConfig | *DiagnosticsStorageAccountConfig   | None             | Copy value   |
+| FabricSettings                    | *[]SettingsSectionDescription    | *[]SettingsSectionDescription      | None             | Copy value   |
+| HttpApplicationGatewayCertificate | *CertificateDescription          |                                    | Property Removed | Property Bag |
+| ManagementEndpoint                | string                           | string                             | None             | Copy value   |
+| NodeTypes                         | []NodeTypes                      | []NodeTypeDescription              | Change of Type   | Property Bag |
+| ReliabilityLevel                  | *Level                           | *ClusterPropertiesReliabilityLevel | Change of Type   | Property Bag |
+| ReverseProxyCertificate           |                                  | *CertificateDescription            | New Property     | Skip         |
+| UpgradeDescription                | *PaasClusterUpgradePolicy        | *ClusterUpgradePolicy              | Change of Type   | Property Bag |
+| UpgradeMode                       |                                  | *ClusterPropertiesUpgradeMode      | New Property     | Skip         |
+| VmImage                           | *string                          | *string                            | None             | Copy value   |
+
+Of the 15 properties listed:
+
+* 8 are unchanged (same name and type); these values are simply copied across.  
```

Comment:
Same comment as above -- they all need autogenerated conversion functions don't they?

Reply:
Why would they need separate functions?

We're going to be generating the central `ConvertTo()` function required by the controller, surely all we need is to include a simple assignment in that function to copy the value from one instance to the next. Yes?

Reply:
See my larger example above -- every single type needs a `ConvertTo` I think?

Reply:
Yes, every custom type needs its own `ConvertTo/From` methods generated.